### DISCUSSION TITLE
[Zephyr 2.4.0] Channel Selection Algorithm 1 (CSA#1) compatibility seems not working.

### DISCUSSION THREAD

**Discussion by JakubWiniarski** [zephyrproject-rtos/zephyr#68720] [Q&A]
I faced some behavior that looks like an error but before reporting it I would like to ask if i'm right. 

1. In zephyr 2.4.0 due it supports BLE 5.0 default Channel Selection Algorithm is CSA#2.
2. According to BLE 5.0 If central see in PDU from device that device does not support CSA#2 then central should use CSA#1
3. My central is using zephyr 2.4.0.
4. I have a device that supports only CSA#1. I see in sniffed ADV_IND frames that device is broadcasting it.
5. But then when central is trying to connect (CONNECT_IND) it uses CSA#2 (I see it in sniffed frames) and it results in connect fail for every connect attempt.
6. I've seen here: 
https://devzone.nordicsemi.com/f/nordic-q-a/96513/change-channel-selection-algorithm-csa-from-default-2-to-1-in-nrf52840dk
suggestion of solution. 
7. When I used it:
```
CONFIG_BT_CTLR_CHAN_SEL_2=n
CONFIG_BT_LL_SW_SPLIT=y
CONFIG_BT_CTLR_ADV_EXT=n
```
I see that central is using only CSA#1 and connect to device is working just fine. But this is just static way to switch central to use  CSA#1 all the time.
 
8. So from my perspective it looks like a problem that zephyr 2.4.0 on central is not handling properly cases when dynamic "rollback" from CSA#2 to CSA#1 is necessary. 
9. I couldn't find any existing error with similar description nor I found anything about that in release notes of next zephyr version.

**Comment by JakubWiniarski** [comment]
Really, no one can help?

**Reply by cvinayak** [reply]
Advertising Extensions was work in progress in the Zephyr 2.4.0. Disabling channel selection #2 is the correction solution. Please try to use latest Zephyr versions and let us know if the issue still exists.
